File: src/main/java/at/hrechny/predictionsbot/exception/interceptor/EnableErrorReport.java
package at.hrechny.predictionsbot.exception.interceptor;

import io.micronaut.aop.Around;
import io.micronaut.context.annotation.Type;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Around
@Type(ExceptionInterceptor.class)
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
public @interface EnableErrorReport {

}

File: src/main/kotlin/at/hrechny/predictionsbot/exception/interceptor/EnableErrorReport.kt
package at.hrechny.predictionsbot.exception.interceptor

import io.micronaut.aop.Around
import io.micronaut.context.annotation.Type

@Around
@Type(ExceptionInterceptor::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
annotation class EnableErrorReport
